src/lib.rs

```diff
@@ -1,6 +1,87 @@
 #![deny(missing_docs)]
 
-//! TODO(nlordell): documentation with examples.
+//! Generate bindings for Ethereum smart contracts. Internally, the generated
+//! types use `web3` crate to interact with the Ethereum network and uses a
+//! custom [`Instance`](ethcontract::contract::Instance) runtime that can be
+//! used directly without code generation.
+//!
+//! This crate is using `std::future::Future` for futures by wrapping `web3`
+//! `futures` 0.1 with `futures` 0.3 compatibility layer. This means that this
+//! crate is ready for `async`/`await`!
+//!
+//! Here is an example of interacing with a smart contract `MyContract`. The
+//! builder pattern is used for configuring transactions.
+//!
+//! ```text
+//! pragma solidity ^0.5.0;
+//!
+//! contract MyContract {
+//!   function my_view_function(uint64 some_val) public view returns (string) {
+//!     // ...
+//!   }
+//!
+//!   function my_function(bool some_bool, string some_str) public returns (uint256) {
+//!     // ...
+//!   }
+//!
+//!   function my_other_function() public {
+//!     // ...
+//!   }
+//! }
+//! ```
+//!
+//! Once this contract is built and deployed with truffle the following example
+//! demonstrates how to interact with it from Rust.
+//!
+//! ```ignore
+//! use ethcontract::transaction::Account;
+//! use std::time::Duration;
+//! use web3::api::Web3;
+//! use web3::types::*;
+//!
+//! // this proc macro generates a `MyContract` type with type-safe bindings to
+//! // contract functions
+//! ethcontract::contract!("path/to/MyContract.json");
+//!
+//! // create a web3 instance as usual
+//! let transport = ...;
```

Comment:
Is this "transport" information less important? the ... makes me wonder what I am missing!

Reply:
@bh2smith it's `web3` related so I omitted it from this example. Its basically how you want to connect to a node.